services/sharepoint-connector/src/msgraph/types/pipeline-item.interface.ts

```diff
@@ -0,0 +1,13 @@
+import { DriveItem, ListItem } from './sharepoint.types';
+
+// This can be named something like ItemToProcess
+export interface PipelineItem {
+  itemType: 'listItem' | 'driveItem';
+  item: DriveItem | ListItem;
+  siteId: string;
+  siteWebUrl: string;
+  driveId: string;
+  driveName: string;
+  folderPath: string;
+  fileName: string;
+}
```

Comment:
Not sure if that helps you with typing, but you can type it more directly 🤔 

[Example on TS playground](https://www.typescriptlang.org/play/?#code/JYOwLgpgTgZghgYwgAgCJWANwgSUgW2QG8AoZc5OALmQHIBnMDEAc1oG4SBfEk0SWIhQAZYIzwRCpCsgBGNEAFd8s6Jx59w0eEmQAhOPVwFiZCvWCQcAExqNmLTjIuQA6hFkBVKABs7TUEczcmsMbBt-BycKUKwIADk4fAhIwOjyGAB7H2toAAU4MAALVNZ05BhgHwSklOR7NO5eCAAPAAdMqDBkMABPNpQ84AGfUGNJZABefUNxwgAyZAAKYNMZGUtJABV+utpY8IIOVY2CGnQ4iXxyrmQAHzX18k38HYGaWlHxI-LTyRpRN9JDcAJS8GCKEAIMDATIgHoQRhLajIIYjMZXEE0TCZYDWR7PGDLOAAOhebxQkypdAOc1oIIJ6wQcPo2QgJJ8mRYyLJBBJcBBN2QEB8RmQwCJPPJuym1M+YjAV3pjJkzJArOqHK5Ur5skFqx4XCAA)

```suggestion
interface BaseItem {
  siteId: string;
  siteWebUrl: string;
  driveId: string;
  driveName: string;
  folderPath: string;
  fileName: string;
}

export type PipelineItem = BaseItem & (
  {
    itemType: 'driveItem';
    item: DriveItem;
  } | {
    itemType: 'listItem';
    item: ListItem;
  })
```